### PR TITLE
Add captcha to sign up

### PR DESCRIPTION
**Pull request by Zimbardoex**
#### Solves:
- #228 

[Code review checklist](https://github.com/op-analytics/NYT-Media-Analytics/wiki/Code-Review-Checklist)

### CODE REVIEW

File: front-end/src/routes/Auth/Signup/Signup.js [hunk 1]
```diff
@@ -4,15 +4,24 @@ import Container from '@material-ui/core/Container';
...
-import React from 'react';
+import MuiAlert from '@material-ui/lab/Alert';
+import React, { useState } from 'react';
+import ReCAPTCHA from 'react-google-recaptcha';
 import { useForm } from 'react-hook-form';
 import { useDispatch } from 'react-redux';
 import { signup } from '../../../state/ducks/user';
 
+
+const CAPTCHA_KEY =
```
Comment (NICHTJ3) [inline front-end/src/routes/Auth/Signup/Signup.js]: The fudge?

File: front-end/src/routes/Auth/Signup/Signup.js [hunk 2]
```diff
@@ -37,10 +46,30 @@ export default function SignUp() {
...
   const submit = ({ name, email, password, confirmPassword }) => {
-    dispatch(signup(name, email, password, confirmPassword));
+    if (isHuman) {
+      dispatch(signup(name, email, password, confirmPassword));
+    } else {
+      setSnackBarOpen(true);
+    }
+  };
+
+  const confirmHuman = () => setIsHuman(true);
```
Comment (NICHTJ3) [inline front-end/src/routes/Auth/Signup/Signup.js]: No need to make a named variable out of this

File: front-end/src/routes/Auth/Signup/Signup.js [hunk 3]
```diff
@@ -37,10 +46,30 @@ export default function SignUp() {
...
+
+  const handleClose = reason => {
+    if (reason === 'clickaway') {
+      return;
+    }
+    setSnackBarOpen(false);
+  };
```
Comment (NICHTJ3) [inline front-end/src/routes/Auth/Signup/Signup.js]: This could be done inline as well

**Comment by NICHTJ3** [comment]
:sparkles:

File: front-end/src/routes/Auth/Signup/Signup.js (hunk 1 above)
Comment (Zimbardoex) [reply front-end/src/routes/Auth/Signup/Signup.js]: Yeah, copy-paste gone wrong :man_shrugging:. Fixed.

File: front-end/src/routes/Auth/Signup/Signup.js (hunk 2 above)
Comment (Zimbardoex) [reply front-end/src/routes/Auth/Signup/Signup.js]: Yup. Right. Fixed.

File: front-end/src/routes/Auth/Signup/Signup.js (hunk 3 above)
Comment (Zimbardoex) [reply front-end/src/routes/Auth/Signup/Signup.js]: Nah, man. Used in both alert and snack bar so made function. 

I have tidied it a little from your suggestion.

**Pull request merged**